ssim/ssimlib.py

```diff
@@ -78,7 +78,11 @@ def ssim_value(self, img2):
         Returns:
           Computed SSIM float value.
         """
-        img2 = SSIMImage(img2, self.gaussian_kernel_1d, self.img1.size)
+        # Performance boost if handed a compatible SSIMImage object.
+        if not isinstance(img2, SSIMImage) \
+          or not (self.gaussian_kernel_1d == img2.gaussian_kernel_1d).all():
```

Comment:
Might this equality check suffer from floating point precision? If so, you could use `numpy.allclose` function.